disnake/auto_moderation.py

```diff
@@ -0,0 +1,495 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2022-present Disnake Development
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+
+from __future__ import annotations
+
+from datetime import timedelta
+from typing import TYPE_CHECKING, List, Literal, Optional, Sequence, Union, overload
+
+from .enums import (
+    AutomodActionType,
+    AutomodEventType,
+    AutomodKeywordPresetType,
+    AutomodTriggerType,
+    enum_if_int,
+    try_enum,
+    try_enum_to_int,
+)
+from .utils import MISSING, _get_as_snowflake
+
+if TYPE_CHECKING:
+    from typing_extensions import Self
+
+    from .abc import Snowflake
+    from .channel import ForumChannel
+    from .guild import Guild, GuildChannel as GuildChannelType, GuildMessageable
+    from .member import Member
+    from .message import Message
+    from .role import Role
+    from .types.auto_moderation import (
+        AutomodAction as AutomodActionPayload,
+        AutomodActionExecutionEvent as AutomodActionExecutionEventPayload,
+        AutomodActionMetadata,
+        AutomodRule as AutomodRulePayload,
+        AutomodTriggerMetadata as AutomodTriggerMetadataPayload,
+        EditAutomodRule as EditAutomodRulePayload,
+    )
+
+__all__ = ("AutomodAction", "AutomodTriggerMetadata", "AutomodRule", "AutomodActionExecution")
```

Comment:
great place for a magic trailing comma

Reply:
c96a7bce1ace4280afa7bd7c1f6cdc07223a3fc4